server/src/main/java/org/elasticsearch/search/aggregations/bucket/filter/FiltersAggregator.java

```diff
@@ -251,7 +251,7 @@ private FiltersAggregator(String name, AggregatorFactories factories, List<Query
     @Override
     public InternalAggregation buildEmptyAggregation() {
         InternalAggregations subAggs = buildEmptySubAggregations();
-        List<InternalFilters.InternalBucket> buckets = new ArrayList<>(filters.size() + otherBucketKey == null ? 0 : 1);
+        List<InternalFilters.InternalBucket> buckets = new ArrayList<>(filters.size() + (otherBucketKey == null ? 0 : 1));
```

Comment:
Nice catch.  I always say, if you have to think about operator precedence, you should add more parenthesis.

Reply:
Truth.